src/main/java/com/couchbase/lite/support/Batcher.java

```diff
@@ -146,7 +146,9 @@ public void flushAll() {
     public void clear() {
         unschedule();
         inbox.clear();
-        inbox.notify();
+        synchronized (inbox) {
```

Comment:
`inbox` is synchronized List. so it might not be necessary. 
`this.inbox = Collections.synchronizedList(new ArrayList<T>());`